lib/src/controls/form/color_picker/color_spectrum.dart

```diff
@@ -0,0 +1,760 @@
+    final backgroundColor = theme.resources.controlSolidFillColorDefault;
+    final textColor = theme.resources.textFillColorPrimary;
+    final borderRadius = BorderRadius.circular(4.0);
+    const labelPadding = EdgeInsets.symmetric(horizontal: 8, vertical: 4);
+
+    final textSpan = TextSpan(
+      text: text,
+      style: TextStyle(color: textColor),
+    );
+
+    final textPainter = TextPainter(
+      text: textSpan,
+      textAlign: TextAlign.left,
+      textDirection: dart.TextDirection
+          .ltr, // TODO: Update if color names support RTL languages in the future.
+    )..layout();
+
+    final labelWidth = textPainter.width + labelPadding.horizontal;
+    final labelHeight = textPainter.height + labelPadding.vertical;
+    final labelX = (position.dx - labelWidth / 2)
+        .clamp(0, size.width - labelWidth)
+        .toDouble();
+    double labelY = (position.dy - labelHeight - 30)
+        .clamp(0, size.height - labelHeight)
+        .toDouble();
+
+    // Check if label would overlap the indicator and adjust position
+    final labelBottomY = labelY + labelHeight;
+    if (position.dy < labelBottomY) {
+      labelY = position.dy + labelHeight - 5;
+    }
+
+    final rect = Rect.fromLTWH(labelX, labelY, labelWidth, labelHeight);
+
+    // Draw background with shadow
+    final shadow = BoxShadow(
+      color: Colors.black.withOpacity(0.1),
+      blurRadius: 4,
+      offset: const Offset(0, 2),
+    );
+
+    // Draw shadow
+    canvas.drawRRect(
+      RRect.fromRectAndCorners(
+        rect.shift(shadow.offset).inflate(shadow.spreadRadius),
+        topLeft: borderRadius.topLeft,
+        topRight: borderRadius.topRight,
+        bottomLeft: borderRadius.bottomLeft,
+        bottomRight: borderRadius.bottomRight,
+      ),
+      Paint()
+        ..color = shadow.color
+        ..maskFilter = MaskFilter.blur(
+          BlurStyle.normal,
+          shadow.blurRadius,
+        ),
+    );
+
+    // Draw background
+    canvas.drawRRect(
+      RRect.fromRectAndCorners(
+        rect,
+        topLeft: borderRadius.topLeft,
+        topRight: borderRadius.topRight,
+        bottomLeft: borderRadius.bottomLeft,
+        bottomRight: borderRadius.bottomRight,
+      ),
+      Paint()..color = backgroundColor,
+    );
+
+    // Draw text
+    textPainter.paint(
+      canvas,
+      Offset(
+        labelX + labelPadding.left,
+        labelY + labelPadding.top,
+      ),
+    );
+  }
+}
+
+/// Custom painter for drawing a checkerboard pattern.
+///
+/// This painter is used to represent transparency in the [ColorPicker]'s alpha slider.
+class CheckerboardPainter extends CustomPainter {
+  @override
+  void paint(Canvas canvas, Size size) {
+    const squareSize = 5;
+    final paint = Paint();
+    final width = size.width.toInt();
+    final height = size.height.toInt();
+
+    for (int i = 0; i < width; i += (squareSize * 2)) {
+      for (int j = 0; j < height; j += (squareSize * 2)) {
```

Comment:
I don't quite understand this one

Reply:
Ah, my intent was drawing two dark squares and two light squares per iteration. But I think I can make it simpler:

```dart
for (int i = 0; i < width; i += squareSize) {
 for (int j = 0; j < height; j += squareSize) {
   // Determine if this position should be a dark square
   final isDarkSquare = (i ~/ squareSize + j ~/ squareSize) % 2 == 0;

   paint.color = isDarkSquare
       ? const Color(0xFFD3D3D3) // Light gray
       : Colors.white;

   canvas.drawRect(
     Rect.fromLTWH(i.toDouble(), j.toDouble(), squareSize.toDouble(),
         squareSize.toDouble()),
     paint,
   );
 }
}